onmt/trainer.py

```diff
@@ -214,14 +241,22 @@ def train(self,
             self.model_saver.save(step)
```

Comment:
The most recent PR also added a new save call outside the loop (because `step == train_steps` may no longer be a valid end condition). Should be updated as well.